opynions/pytest_opynions.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+"""
+pytest plugin to test if specific repo follows standards
+"""
+import os
+from collections import defaultdict
+
+import pytest
+
+import yaml
+
+session_data_holder_dict = defaultdict(dict)
+
+@pytest.fixture(scope="session")
+def all_results():
+    return session_data_holder_dict
+
+def pytest_configure(config):
+    """
+    pytest hook used to add pytest_opynions install dir as place for pytest to find tests
+    """
+
+    if config.getoption("repo_health_check"):
+
+        # Add path to pytest-opynions dir so pytest knows where to look for checks
+        file_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+        config.args.append(file_dir)
+
+        # add repo path so a repo can design their own checks inside a repo_state_checks dir
+        repo_path = config.getoption("repo_path")
+        config.args.append(repo_path)
+
+        # Change test prefix to check
+        config._inicache['python_files'] = ['check_*.py']  # pylint: disable=protected-access
+        config._inicache['python_functions'] = ['check_*']  # pylint: disable=protected-access
+    return config
+
+
+def pytest_addoption(parser):
+    """
+    pytest hook used to add command line options used by this plugin
+    """
+    group = parser.getgroup('opynions')
+    group.addoption(
+        "--repo-path",
+        action="store",
+        dest='repo_path',
+        default=None,
+        help="path of repo on which to perform tests"
+    )
+
+    # Since pytest opynions modifies many pytest setting, this flag is necessary
+    # to make sure pytest settings are only changed when health check is suppose to happen
+    group.addoption(
+        "--repo-health-check",
```

Comment:
Maybe use something like "--opynions" or "--opynions-mode" here instead?

Reply:
going to go with --opynions